test/puppetlabs/trapperkeeper/optional_deps_test.clj

```diff
@@ -32,6 +32,83 @@
                                             couplet))))
 
 (deftest optional-deps-test
+  (testing "when using defservice"
+    (testing "with a protocol"
+      (defservice poetry-svc-w-protocol
+        PoetryService
+        {:required [HaikuService]
+         :optional [SonnetService]}
+        (init [this ctx] ctx)
+        (get-haiku [this] "haiku")
+        (get-sonnet [this] "sonnet"))
```

Comment:
This is a nitpick and a stylistic thing that I won't mind if I'm overruled on, but, I think it's weird to call `def*` things inside of other forms.  So I'd probably write these tests such that the `defservice` was at the top level of the file, just prior to the test that was consuming it.